lib/cache/collections.go

```diff
@@ -1246,9 +1246,20 @@ func (databaseExecutor) getAll(ctx context.Context, cache *Cache, loadSecrets bo
 
 func (databaseExecutor) upsert(ctx context.Context, cache *Cache, resource types.Database) error {
 	if err := cache.databasesCache.CreateDatabase(ctx, resource); err != nil {
+		// Skip databases that failed to be validated to avoid stopping cache
+		// backend events consumption (which can lead to an invalid cache
+		// state). This case can happen during Teleport upgrades/downgrades,
+		// where the database stored on the backend is no longer valid given the
+		// Teleport version.
+		if trace.IsBadParameter(err) {
+			cache.Logger.Infof("Skipping database %q addition to cache due to validation errors. Solve the errors and update your database spec to get it on cache: %s", resource.GetName(), err)
```

Comment:
I wonder if we could say something like
"Adding database %q to cache was skipped due to validation errors. Please update your database spec accordingly: %s"

I'm not a wordsmith so feel free to ignore above, but I think we can smooth out the error messaging a bit.

Reply:
Thanks for the suggestion.